test_app/tests/rbac/test_triggers.py

```
@@ -1005,3 +1005,39 @@ def test_remove_permission_team_query_count(self, team, inventory, inv_rd):
+
+        def tracking_descendent(self):
+            called_on_pks.append(self.pk)
+            return original_descendent(self)
+
+        with mock_patch.object(ObjectRole, 'descendent_roles', tracking_descendent):
+            inv_rd.give_permission(team, inventory)
+
+        for pk in called_on_pks:
+            assert pk not in ancestor_pks, (
+                f"descendent_roles() was called on ancestor ObjectRole pk={pk}; "
+                f"this causes O(ancestors × descendants) recomputation"
```

Ruff (0.16.1)

[warning] String contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF001)